kong/plugins/jwt/handler.lua

```diff
@@ -175,8 +175,9 @@ local function do_authentication(conf)
     return false, {status = 401, message = "Invalid algorithm"}
   end
 
-  local jwt_secret_value = algorithm ~= nil and algorithm:sub(1, 2) == "HS" and
-                           jwt_secret.secret or jwt_secret.rsa_public_key
+  local jwt_secret_value = algorithm ~= nil and algorithm == "none" or
+                           algorithm ~= nil and algorithm:sub(1, 2) == "HS" and jwt_secret.secret or
+                           jwt_secret.rsa_public_key
```

Comment:
this is becoming very hard to read. Maybe we can change this into some if-then statements to make it easier to read/maintain